privacyidea/lib/utils.py

```diff
@@ -1158,3 +1158,54 @@ def truncate_comma_list(data, max_len):
             new_data.append(d)
         data = new_data
     return ",".join(data)
+
+
+def check_pin_policy(pin, policy):
+    """
+    The policy to check a PIN can contain of "c", "n" and "s".
+    "cn" means, that the PIN should contain a character and a number.
+    "+cn" means, that the PIN should contain elements from the group of characters and numbers
+    "-ns" means, that the PIN must not contain numbers or special characters
+
+    :param pin: The PIN to check
+    :param policy: The policy that describes the allowed contents of the PIN.
+    :return: Tuple of True or False and a description
+    """
+    chars = {"c": "[a-zA-Z]",
+             "n": "[0-9]",
+             "s": "[.:,;_<>+*!/()=?$§%&#~\^-]"}
```

Comment:
Should this include [ and ]? :-)

Reply:
No. This would change the current behaviour ;-)
Lets do this in another PR, please. (I know I am not consequent here)